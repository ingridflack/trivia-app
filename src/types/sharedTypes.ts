export interface Category {
  id: number;
  name: string;
}

export interface ApiResponse {
  message: string;
}

interface User {
  _id: string;
  mame: string;
  username: string;
  avatar: string;
}

export interface Question {
  category: string;
  difficulty: string;
  question: string;
  answers: string[];
  _id: string;
}
export interface Trivia {
  category: string;
  createdAt: string;
  difficulty: string;
  game_mode: string;
  questions: Question[];
  score: number;
  users: User[];
  _id: string;
}

export interface TriviaHistory {
  items: {
    answerTime: number;
    isCorrect: boolean;
    question: Question;
    _id: string;
  }[];
  trivia: Trivia;
}
