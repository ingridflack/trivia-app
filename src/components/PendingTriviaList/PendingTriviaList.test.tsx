import { render } from "@testing-library/react";

import { describe, it, expect } from "vitest";
import { PendingTriviaList } from ".";
import { PendingTrivia } from "../../types/sharedTypes";

const mockPendingTriviaList: PendingTrivia[] = [
  {
    trivia: {
      category: "science",
      difficulty: "easy",
      users: [
        { username: "User1", _id: "1" },
        { username: "User2", _id: "2", avatar: "avatar1.png" },
      ],
      _id: "66c8f351942dd47ad29d4ba1",
    },
  },
  {
    trivia: {
      category: "history",
      difficulty: "medium",
      users: [
        { username: "User3", _id: "3" },
        { username: "User4", _id: "4", avatar: "avatar2.png" },
      ],
      _id: "66c8f351942dd47ad29d4ba2",
    },
  },
];

describe("PendingTriviaList", () => {
  it("should match the snapshot", () => {
    const { container } = render(
      <PendingTriviaList pendingTriviaList={mockPendingTriviaList} />
    );

    expect(container).toMatchSnapshot();
  });

  it("should render the pending trivia list with the correct amount of items", () => {
    const { container } = render(
      <PendingTriviaList pendingTriviaList={mockPendingTriviaList} />
    );

    const pendingTriviaItems = container.querySelectorAll(".chakra-card");

    expect(pendingTriviaItems.length).toBe(mockPendingTriviaList.length);
  });
});
