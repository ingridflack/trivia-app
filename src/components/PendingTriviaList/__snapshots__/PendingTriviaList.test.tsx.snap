// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PendingTriviaList > should match the snapshot 1`] = `
<div>
  <div
    class="chakra-container css-8f57v9"
  >
    <p
      class="chakra-text css-12a0wam"
    >
      Pending Trivia
    </p>
    <div
      class="chakra-stack css-17gphto"
    >
      <div
        class="chakra-card css-u646wo"
        role="group"
      >
        <div
          class="chakra-stack css-1y3f6ad"
        >
          <p
            class="chakra-text css-ax9bdp"
          >
            General Knowledge
          </p>
          <span
            class="chakra-badge css-ifsvay"
          >
            easy
          </span>
        </div>
        <div
          class="chakra-avatar__group css-1fv303w"
          role="group"
        >
          <span
            class="chakra-avatar css-1xczbvk"
          >
            <div
              aria-label="User2"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              U
            </div>
          </span>
          <span
            class="chakra-avatar css-1o6hphw"
          >
            <div
              aria-label="User1"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              U
            </div>
          </span>
        </div>
        <div
          class="chakra-stack css-10cxrny"
        >
          <a
            class="chakra-button css-4xx2wk"
            href="/trivia/66c8f351942dd47ad29d4ba1"
          >
            Play!
          </a>
        </div>
      </div>
      <div
        class="chakra-card css-u646wo"
        role="group"
      >
        <div
          class="chakra-stack css-1y3f6ad"
        >
          <p
            class="chakra-text css-ax9bdp"
          >
            General Knowledge
          </p>
          <span
            class="chakra-badge css-ifsvay"
          >
            medium
          </span>
        </div>
        <div
          class="chakra-avatar__group css-1fv303w"
          role="group"
        >
          <span
            class="chakra-avatar css-1xczbvk"
          >
            <div
              aria-label="User4"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              U
            </div>
          </span>
          <span
            class="chakra-avatar css-1o6hphw"
          >
            <div
              aria-label="User3"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              U
            </div>
          </span>
        </div>
        <div
          class="chakra-stack css-10cxrny"
        >
          <a
            class="chakra-button css-4xx2wk"
            href="/trivia/66c8f351942dd47ad29d4ba2"
          >
            Play!
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
